docs/guides/implementation-guide.mdx

```diff
@@ -361,22 +408,20 @@ then the message will be kicked to an exponential backoff retry queue. The relay
 
 The Relayer may also require gas payment for a specific message ID on the origin chain before processing the message on the destination chain. To do this, they must have an IGP deployed with their address set as beneficiary and index gas payment events. See [gas payment enforcement trait](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/agents/relayer/src/msg/gas_payment/mod.rs). We recommend to start with no gas payment enforcement policy and then gradually support more restrictive ones.
 
-:::tip
+### Testing
 
-Once you implemented the MVP of the relayer, you can create an end-to-end test that
+#### End-to-end
 
-1. Spins up local origin and destination chain
-2. Deploys your contracts onto both
-3. Run validators for the origin chain
-4. Run a relayer between both chains
-5. Observe that upon dispatch of a message of the origin chain, the validator observes the message, creates a signature and the relayer appropriately processes your message via the ISM that specifies the validator on the destination chain.
+Once you have implemented an MVP of the relayer, you should create an end-to-end test that:
 
+1. Spins up local origin and destination chains.
+2. Deploys your contracts onto both chains.
+3. Run validators for the origin chain.
+4. Run a relayer between both chains.
+5. Observe that upon dispatch of a message of the origin chain, the validator observes the message, creates a signature and the relayer appropriately processes your message via the ISM that specifies the validator on the destination chain.
 
 See this [end-to-end test on the Rust codebase ](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/rust/utils/run-locally/src/main.rs)for inspiration.
 
```

Comment:
from ticket:
> one definite change in v3 is we added the requirement for e2e test + remote testnet test

not sure what exactly this remote testnet test is, will sync

Reply:
update: it means adding some text/instructions around making sure it works on a testnet